src/utils/streams/promise_from_streams.js

```diff
@@ -0,0 +1,67 @@
+/**
+ *  Take an array of streams, pipe the output
+ *  from each one into the next, listening for
+ *  errors from any of the streams, and then resolve
+ *  the promise once the final stream has finished
+ *  writing/reading.
+ *
+ *  If the last stream is readable, it's final value
+ *  will be provided as the promise value.
```

Comment:
That said, I agree that outside of tests and systems that use streams exclusively, for everything, this feature has limited value.